config/v1/types_authentication.go

```diff
@@ -192,6 +197,14 @@ type OIDCProvider struct {
 	// +required
 	Issuer TokenIssuer `json:"issuer"`
 
+	// OIDCClients contains configuration for the platform's clients that
+	// need to request tokens from the issuer
+	//
+	// +listType=map
+	// +listMapKey=componentNamespace
+	// +listMapKey=componentName
+	OIDCClients []OIDCClientConfig `json:"oidcClients"`
```

Comment:
My gut says it's possible to write a CEL expression at the `Authentication` level that requires items in the spec list to either

Reply:
my first attempt is apparently only 50.3x more complex than the rule cost allows us if we limit all OIDCClients to max 20
```
self.spec.oidcProviders.all(
  p, p.oidcClients.all(
    specC, self.status.oidcClients.exists(
      statusC, statusC.componentNamespace == specC.componentNamespace && statusC.componentName == specC.componentName)
      || oldSelf.spec.oidcProviders.exists(
            oldP, oldP.oidcClients.exists(
            oldC, oldC.componentNamespace == specC.componentNamespace && oldC.componentName == specC.componentName))))
```
I'll try to see if we can optimize it somehow but I'm not sure we'd fit within the rule cost budget once we allow multiple OIDC providers anyway :neutral_face:

Reply:
I was hoping that we could perhaps try to use the fact that the arrays are meant to be accessed as maps but as I originally suspected, that's apparently only useful for SSA.